src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPTextViewConnectionListener.cs

```diff
@@ -122,34 +118,45 @@ public void SubjectBuffersConnected(ITextView textView, ConnectionReason reason,
 
             RazorLSPTextViewFilter.CreateAndRegister(vsTextView);
 
-            lock (_lock)
+            if (textView.TextBuffer.IsRazorLSPBuffer())
```

Comment:
nit: In the future to avoid extra nesting a nice pattern I typically use is:

```C#
if (!textView.TextBuffer.IsRazorLSPBuffer())
{
    return;
}
```

That way less nesting, and once you pass a certain execution point it's less to keep in your head as to what's happening. Even though it's more verbose I've found the expressiveness of it tends to be a life saver at debug / code readability time

Reply:
Whoops you're right! Fixed.